test/TEST-04-FULL-SYSTEMD/test.sh

```diff
@@ -54,79 +52,33 @@ test_setup() {
-        ln -sfn /run/lock "$initdir/var/lock"
-
-        inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip \
-            mount dmesg mkdir cp ping dd \
-            umount strace less setsid systemctl reset sync
-
-        for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
-            [ -f ${_terminfodir}/l/linux ] && break
-        done
-        inst_multiple -o ${_terminfodir}/l/linux
-        inst_multiple grep
-        inst_simple ./fstab /etc/fstab
-        if type -P rpm &> /dev/null; then
-            rpm -ql systemd | xargs -r "$DRACUT_INSTALL" ${initdir:+-D "$initdir"} -o -a -l
-        elif type -P dpkg &> /dev/null; then
-            dpkg -L systemd | xargs -r "$DRACUT_INSTALL" ${initdir:+-D "$initdir"} -o -a -l
-        elif type -P pacman &> /dev/null; then
-            pacman -Q -l systemd | while read -r _ a; do printf -- "%s\0" "$a"; done | xargs -0 -r "$DRACUT_INSTALL" ${initdir:+-D "$initdir"} -o -a -l
-            rm "$initdir"/usr/lib/systemd/system/sysinit.target.wants/systemd-firstboot.service
-        else
-            echo "Can't install systemd base"
-            return 1
-        fi
-        inst /sbin/init
-        inst_multiple -o {,/usr}/lib/systemd/system/"dracut*"
-
-        inst_simple "${basedir}/modules.d/99base/dracut-lib.sh" "/lib/dracut-lib.sh"
-        inst_simple "${basedir}/modules.d/99base/dracut-dev-lib.sh" "/lib/dracut-dev-lib.sh"
-        inst_binary "${basedir}/dracut-util" "/usr/bin/dracut-util"
-        ln -s dracut-util "${initdir}/usr/bin/dracut-getarg"
-        ln -s dracut-util "${initdir}/usr/bin/dracut-getargs"
-
-        # make a journal directory
-        mkdir -p "$initdir"/var/log/journal
-
-        # install some basic config files
-        inst_multiple -o \
-            /etc/machine-id \
-            /etc/adjtime \
-            /etc/passwd \
-            /etc/shadow \
-            /etc/group \
-            /etc/shells \
-            {,/usr}/etc/nsswitch.conf \
-            /etc/pam.conf \
-            /etc/securetty \
-            /etc/os-release \
-            /etc/localtime
-
-        # we want an empty environment
-        : > "$initdir"/etc/environment
+    "$basedir"/dracut.sh -l --keep --tmpdir "$TESTDIR" \
+        -m "test-root" \
+        -I "ldconfig" \
+        -i ./test-init.sh /sbin/test-init \
+        -i ./fstab /etc/fstab \
+        -i "${basedir}/modules.d/99base/dracut-lib.sh" "/lib/dracut-lib.sh" \
+        -i "${basedir}/modules.d/99base/dracut-dev-lib.sh" "/lib/dracut-dev-lib.sh" \
+        --no-hostonly --no-hostonly-cmdline --nomdadmconf --nohardlink \
+        -f "$TESTDIR"/initramfs.root "$KVERSION" || return 1
+
+    mkdir -p "$TESTDIR"/overlay/source && cp -a "$TESTDIR"/dracut.*/initramfs/* "$TESTDIR"/overlay/source && rm -rf "$TESTDIR"/dracut.* && export initdir=$TESTDIR/overlay/source
+
+    if type -P rpm &> /dev/null; then
+        rpm -ql systemd | xargs -r "$basedir"/dracut-install ${initdir:+-D "$initdir"} -o -a -l
```

Comment:
I'm not sure if we should continue testing systemd like this, this is installing and enabling all units, even some of them are not included by any dracut systemd module (that's why some tweaks are being made in the old test from line 205 to 235, missing with new changes). Maybe we can change the approach of this test, and do not install only the systemd module, but try to install all the dracut systemd modules (line 149).

Reply:
My goal was to keep the test simple and maintainable. The test as is already passing on all target containers. Would adding more code make the test more maintainable ? Or would it increase test coverage for _dracut_ by adding more code ? 

If I understand your suggestion correctly, you proposed to add more code to make initramfs.root more complete, but would that actually test dracut itself, or we would just end up testing systemd itself on the target container ? As I understand the main goal of these tests is to test initramfs.testing and not initramfs.root.
That said, I do not have a strong opinion on this and happy to add back the code from old test from line 205 to 235.

> try to install all the dracut systemd modules (line 149).
I think this is a good idea, but this is a suggestion for another PR. This is a suggestion to change the scope of this test case, and this PR is already complex enough just trying to do the refactoring.

Reply:
> add back the code from old test from line 205 to 235.

I convinced myself to make this change now as the goal of this PR not to optimize, but just refactor.
It seems this additional piece of code only picks up the following 2 additional files:
- /sbin/agetty
- /usr/lib/systemd/systemd-random-seed